src/ace/managers/game.c

```diff
@@ -5,108 +5,128 @@
 #include <ace/managers/game.h>
 #include <ace/managers/system.h> // systemKill
 
-/* Globals */
-tGameManager g_sGameManager;
+UBYTE s_isGameRunning = 1;
 
 /* Functions */
-void gameCreate(void) {
-	logBlockBegin("gameCreate()");
-	g_sGameManager.ubStateCount = 0;
-	g_sGameManager.pStateFirst = 0;
-	// Here or in push or else game will quit instantly.
-	// I guess it should stay here, makes each push a bit faster
-	g_sGameManager.isRunning = 1;
-	logBlockEnd("gameCreate()");
+void gameExit(void) {
+	s_isGameRunning = 0;
 }
 
 UBYTE gameIsRunning(void) {
-	return g_sGameManager.isRunning;
+	return s_isGameRunning;
```

Comment:
I have a problem with this variable.

- it doesn't get set on gamestate manager creation
- it doesn't get set/unset when you push/pop states so your game must call gameExit at some point.

So if you're just pushing and popping your states, gameIsRunning doesn't change to 0 by itself so the game won't close automatically if you just check if game is running like in the old way.

Reply:
As discussed offline, variable is not actually needed, as same kind of logic could be archived by checking if current state is not null.